march_joint_inertia_controller/src/inertia_estimator.cpp

```diff
@@ -0,0 +1,290 @@
+    //            (joint_torque_[0] * sos_[i][3] + joint_torque_[1] * sos_[i][4] + joint_torque_[2] * sos_[i][5]);
+  }
+  it = filtered_joint_torque_.begin();
+  it = filtered_joint_torque_.insert(it, x);
+  filtered_joint_torque_.resize(fil_tor_size_);
+
+  // temp = 1.0;
+}
+
+// Estimate the inertia using the acceleration and torque
+void InertiaEstimator::inertia_estimate()
+{
+  double torque_e;
+  double acc_e;
+  apply_butter();
+
+  correlation_calculation();
+  K_i_ = gain_calculation();
+  K_a_ = alpha_calculation();
+  torque_e = filtered_joint_torque_[0] - filtered_joint_torque_[1];
+  acc_e = filtered_acceleration_array_[0] - filtered_acceleration_array_[1];
+  joint_inertia_ = (torque_e - (acc_e * joint_inertia_)) * K_i_ * K_a_ + joint_inertia_;
+}
+
+// Calculate the alpha coefficient for the inertia estimate
+double InertiaEstimator::alpha_calculation()
+{
+  double vib = vibration_calculation();
+  if (vib < min_alpha_)
+  {
+    vib = min_alpha_;
+  }
+  else if (vib > max_alpha_)
+  {
+    vib = max_alpha_;
+  }
+  return (vib - min_alpha_) / (max_alpha_ - min_alpha_);
+}
+
+// Calculate the inertia gain for the inertia estimate
+double InertiaEstimator::gain_calculation()
+{
+  return (corr_coeff_ * (filtered_acceleration_array_[0] - filtered_acceleration_array_[1])) /
+         (lambda_ + corr_coeff_ * pow(filtered_acceleration_array_[0] - filtered_acceleration_array_[1], 2));
+}
+
+// Calculate the correlation coefficient of the acceleration buffer
+void InertiaEstimator::correlation_calculation()
+{
+  corr_coeff_ =
+      corr_coeff_ / (lambda_ + corr_coeff_ * pow(filtered_acceleration_array_[0] - filtered_acceleration_array_[1], 2));
+  double large_number = pow(10, 8);
+  if (corr_coeff_ > large_number)
+  {
+    corr_coeff_ = large_number;
+  }
+}
+
+// Calculate the vibration based on the acceleration
+double InertiaEstimator::vibration_calculation()
+{
+  std::vector<double> b;
+  b.resize(acc_size_);
+  absolute(filtered_acceleration_array_, b);
+  // moa = mean of the absolute
+  moa_ = mean(b);
+  // aom = absolute of the mean
+  aom_ = abs(mean(filtered_acceleration_array_));
+  // Divide by zero protection, necessary when exo hasn't homed yet
+  if (aom_ == 0.0)
+  {
+    return 0.0;
+  }
+  return moa_ / aom_;
+}
+
+// Calculate a discrete derivative of the speed measurements
+void InertiaEstimator::discrete_speed_derivative(double velocity, const ros::Duration& period)
+{
+  auto it = acceleration_array_.begin();
+  it = acceleration_array_.insert(it, (velocity - velocity_array_[1]) / period.toSec());
+}
+
+void InertiaEstimator::init_p(unsigned int samples)
+{
+  // Setup the initial value for the correlation coefficient 100*standarddeviation(acceleration)^2
+  double mean_value = mean(standard_deviation);
+  double sum = 0;
+  for (unsigned int i = 0; i < standard_deviation.size(); ++i)
+  {
+    sum += pow((standard_deviation[i] - mean_value), 2);
+  }
+  corr_coeff_ = 100 * (sum / samples);
+  return;
```

Comment:
`return` is not necessary for `void` methods.

Reply:
I don't know why you still haven't removed this, but just a friendly reminder :wink: